tests/tt_eager/ops/test_average_pool.cpp

```diff
@@ -34,8 +34,8 @@ int main () {
     tt::tt_metal::LegacyShape resnet18_shape = {1, 1, 7 * 7, 2048};
     auto result = run_avg_pool_2d_resnet(resnet18_shape, device);
 
-    TT_FATAL(result.get_legacy_shape() == tt::tt_metal::LegacyShape({1, 1, tt::constants::TILE_HEIGHT, 2048}), "Incorrect shape {}.", result.get_legacy_shape());
-    TT_FATAL(result.get_legacy_shape().without_padding() == tt::tt_metal::LegacyShape({1, 1, 1, 2048}), "Incorrect shape {}.", result.get_legacy_shape().without_padding());
+    TT_FATAL(result.get_padded_shape() == ttnn::SimpleShape({1, 1, tt::constants::TILE_HEIGHT, 2048}), "Incorrect shape {}.", result.get_padded_shape());
```

Comment:
Is the long term plan to have 
- two shape objects, one padded and one logical; or
- a single shape object, that contains both logical and padded information?

I'm wondering because there will be different trade offs.

Reply:
Shape will just be a vector with rank. It should be used to represent just logical shape, but users can also use it to represent padded shapes if they want. All padding information will be owned by tensor + (new) layout